### PR TITLE
[5.1] Extract resetting user's password to method

### PR DESCRIPTION
**Pull request by alexsoft**
Having such in ResetsPasswords trait, it makes possible to override only passwordReset method to change the way new password for user is set.

### CODE REVIEW

File: src/Illuminate/Foundation/Auth/ResetsPasswords.php [hunk 1]
```diff
@@ -89,11 +89,7 @@ public function postReset(Request $request)
 
 		$response = Password::reset($credentials, function($user, $password)
 		{
-			$user->password = bcrypt($password);
-
-			$user->save();
-
-			Auth::login($user);
+			call_user_func([$this, 'passwordReset'], $user, $password);
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Auth/ResetsPasswords.php]: no need to call_user_fund

File: src/Illuminate/Foundation/Auth/ResetsPasswords.php [hunk 2]
```diff
@@ -109,6 +105,21 @@ public function postReset(Request $request)
 	}
 
 	/**
+	 * Actually reset user's password
+	 *
+	 * @param $user
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Auth/ResetsPasswords.php]: you need to document the types

**Comment by taylorotwell** [comment]
why not just `resetPassword` instead of having it backwards as `passwordReset`?

**Comment by alexsoft** [comment]
@taylorotwell you want me to change it, don't you?

**Pull request merged**